src/bin/pg_autoctl/keeper_pg_init.c

```diff
@@ -774,6 +758,12 @@ create_database_and_extension(Keeper *keeper)
 		 * Connect to pgsql as the system user to create extension: Same
 		 * user as initdb with superuser privileges.
 		 */
+		if (!keeper_update_pg_state(keeper))
```

Comment:
Let's move this before the CREATE EXTENSION log line 5 lines above to keep clear logging. Also let's add a comment why this call is needed, because it's not immediately clear to me. What's the reason this is only needed for Citus nodes?

Reply:
We need to re-init our connection string to do the `create extension` as the proper user, and calling that function is doing that for us, IIRC. I added a comment to that end.